trie-db/src/triedbmut.rs

```diff
@@ -430,21 +431,21 @@ where
 	}
 
 	/// insert a key-value pair into the trie, creating new nodes if necessary.
-	fn insert_at(&mut self, handle: NodeHandle<H::Out>, partial: NibbleSlice, value: DBValue, old_val: &mut Option<DBValue>) -> Result<(StorageHandle, bool), H::Out, C::Error> {
+	fn insert_at(&mut self, handle: NodeHandle<H::Out>, partial: NibbleSlice, key: NodeKey, value: DBValue, old_val: &mut Option<DBValue>) -> Result<(StorageHandle, bool), H::Out, C::Error> {
```

Comment:
In trie_db_mut.rs ,  function arguments  `partial: NibbleSlice, key:NodeKey` could be fused in a single struct (operation on both seems symetric: key increase from what partial decrease on recursive calls at some/many place).
eg
```rust
let child_key = combine_encoded(&key, &partial.encoded_leftmost(cp, false));
match self.remove_at(child_branch, partial.mid(cp), child_key.clone(), old_val)? {
```
do not know if it is worth a new struct, just mentioning it.